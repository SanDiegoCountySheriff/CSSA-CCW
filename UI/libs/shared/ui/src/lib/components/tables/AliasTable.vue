<template>
  <v-container fluid>
    <v-simple-table class="my-3">
      <template #default>
        <thead>
          <tr>
            <th scope="col">{{ $t('Previous Last name') }}</th>
            <th scope="col">{{ $t('Previous First name') }}</th>
            <th scope="col">{{ $t('Previous Middle name') }}</th>
            <th scope="col">{{ $t('City where changed') }}</th>
            <th scope="col">{{ $t('State or region where changed') }}</th>
            <th scope="col">{{ $t('Court file number') }}</th>
            <th scope="col"></th>
          </tr>
        </thead>
        <tbody>
          <tr
            v-for="(alias, index) in aliases"
            :key="index"
          >
            <td>
              {{ alias.prevLastName }}
            </td>
            <td>
              {{ alias.prevFirstName }}
            </td>
            <td>
              {{ alias.prevMiddleName }}
            </td>
            <td>
              {{ alias.cityWhereChanged }}
            </td>
            <td>
              {{ alias.stateWhereChanged }}
            </td>
            <td>
              {{ alias.courtFileNumber }}
            </td>
            <td>
              <v-tooltip bottom>
                <template #activator="{ on, attrs }">
                  <v-btn
                    text
                    color="error"
                    @click="handleDelete(index)"
                    v-bind="attrs"
                    v-on="on"
                  >
                    <v-icon color="error"> mdi-close-circle </v-icon>
                  </v-btn>
                </template>
                {{ $t('Delete item') }}
              </v-tooltip>
            </td>
          </tr>
        </tbody>
      </template>
    </v-simple-table>
  </v-container>
</template>

<script setup lang="ts">
import { AliasType } from '@shared-utils/types/defaultTypes';

interface AliasTableProps {
  aliases?: Array<AliasType>;
}

const emit = defineEmits(['delete']);

const props = withDefaults(defineProps<AliasTableProps>(), {
  aliases: () => [],
});

function handleDelete(index) {
  emit('delete', index);
}
</script>
