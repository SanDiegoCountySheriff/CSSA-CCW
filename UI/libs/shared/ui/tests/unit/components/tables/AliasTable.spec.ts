/* eslint-disable @typescript-eslint/ban-ts-comment */
import AliasTable from '@shared-ui/components/tables/AliasTable.vue';
import Vue from 'vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const alias = {
  prevLastName: 'pLast',
  prevFirstName: 'pFirst',
  prevMiddleName: 'pMiddle',
  cityWhereChanged: 'city',
  stateWhereChanged: 'state',
  courtFileNumber: 'court',
};

describe('AliasTable', () => {
  let vuetify;
  const tMock = {
    $t: text => text,
  };

  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });

  it('should match the snapshot', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      mocks: tMock,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render the Alias table with the correct text in the first row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(0).element.textContent).toContain('pLast');
  });

  it('Should render the Alias table with the correct text in the second row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(1).element.textContent).toContain('pFirst');
  });

  it('Should render the Alias table with the correct text in the third row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(2).element.textContent).toContain(
      'pMiddle'
    );
  });
  it('Should render the Alias table with the correct text in the fourth row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(3).element.textContent).toContain('city');
  });
  it('Should render the Alias table with the correct text in the fifth row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(4).element.textContent).toContain('state');
  });
  it('Should render the Alias table with the correct text in the sixth row', () => {
    //@ts-ignore
    const wrapper = mount(AliasTable, {
      localVue,
      vuetify,
      propsData: {
        aliases: [alias],
      },
      mocks: tMock,
    });

    expect(wrapper.findAll('td').at(5).element.textContent).toContain('court');
  });
});
