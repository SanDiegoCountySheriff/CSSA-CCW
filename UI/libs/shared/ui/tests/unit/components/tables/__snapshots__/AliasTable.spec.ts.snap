// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AliasTable should match the snapshot 1`] = `
<div class="container container--fluid">
  <div class="v-data-table my-3 theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <thead>
          <tr>
            <th>Previous Last name</th>
            <th>Previous First name</th>
            <th>Previous Middle name</th>
            <th>City where changed</th>
            <th>State or region where changed</th>
            <th>Court file number</th>
            <th></th>
          </tr>
        </thead>
        <tbody></tbody>
      </table>
    </div>
  </div>
</div>
`;
